### ISSUE TITLE
nl_after_annotation bug

### ISSUE THREAD

**Issue by NvrBst** [uncrustify/uncrustify#1845]
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastbin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

Debug File:
```
# Uncrustify-0.67-87-d75a44aa9
nl_after_annotation             = force
```
Input (Java):
```
public class Version implements Comparable<Version> {

    @Override
    public int compareTo(@NonNull Version that) {
        return 0;
    }
}
```
Output (Java):
```
public class Version implements Comparable<Version> {

    @Override
    public int compareTo(@NonNull
                         Version that) {
	return 0;
    }
}
```

Based on the description, it shouldn't be adding a new line after the "@NonNull" since it isn't after a new line:
```
nl_after_annotation: Whether to put a newline after a Java annotation statement.
Only affects annotations that are after a newline.
```

Basically I want the function annotations on their own line, but, I want the parameter annotations on a single line is what I was trying for. It works fine for the @Override (even if I put it on the same line, it'll correctly add the new line).
I did my testing with: https://cdanu.github.io/uncrustify_config_preview/index.html

**Comment by guy-maurel** [comment]
PR #2243 is a bugfix proposal.
Please test it and report.
